internal/gps/solver.go

```diff
@@ -494,11 +522,14 @@ func (s *solver) solve() (map[atom]map[string]struct{}, error) {
 			if err != nil {
 				// Err means a failure somewhere down the line; try backtracking.
 				s.traceStartBacktrack(bmi, err, false)
-				s.mtr.pop()
-				if s.backtrack() {
+				success, berr := s.backtrack(ctx)
+				if berr != nil {
+					err = berr
+				} else if success {
 					// backtracking succeeded, move to the next unselected id
 					continue
 				}
+				s.mtr.pop()
```

Comment:
Is the reordering of `pop()` and `backtrack()` significant? It looks like this aligns with another usage now.

Reply:
it is, though actually, the one i changed was right. the one that pops _after_ backtracking was wrong, and has been for some time. fixing that in a follow-up commit.